module-2/backend/week_two.md

```diff
@@ -26,20 +30,59 @@ end
 
 Now how would you find the owner of the team with an id of 4?
 
+> `Team.first.owner`
+
 5. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+
+>Teacher `has_many` students and Student `has_and_belongs_to_many` teachers or just `belongs_to` Teacher depending on the school district I guess.
+
+
 6. Define foreign key, primary key, and schema.
+
+> foreign key is how one table accesses the relationship to another table. The primary key is the unique identifier of that specific table. Schema is the diagram of how those table are connected, through the foerign keys/ primary keys.
+
 7. Describe the relationship between a foreign key on one table and a primary key on another table.
+
+>it's how the tables access the relationships, ie `has_many`/`belongs_to`
+
 8. What are the parts of an HTTP response?
 
+> The verbs. `GET POST PUT DELETE` The `GET` request responds to a few actions. It will take one from the landing page, the index page, and the show page. For `POST PUT & DELET` there are two steps. First, the `GET` request for the specific `id` and then the corresponding verb for whatever the client has requested. `POST` for new content, `PUT` for updating content and `DELETE` for destroying content
```

Comment:
These verbs are part of the **request**.

A response has:

* status line (including a status code)
* headers
* body